src/main/java/me/eccentric_nz/TARDIS/listeners/TARDISItemFrameListener.java

```diff
@@ -387,7 +387,8 @@ public void onItemFrameDamage(EntityDamageByEntityEvent event) {
                         }, 40L);
                     } else {
                         // is the handles currently talking?
-                        if (talkingHandles.stream().anyMatch(id -> id.equals(handlesId))) { // match by predicate since handlesId was fetched from database again
+                        // match by predicate in case multiple entries are in list. can happen when handles is clicked many times repeatedly
```

Comment:
Can we not use a Set instead of a List, then the id can only ever be in there once?

Reply:
I believe the handles ID is technically a different object reference each time he's clicked, since I'm casting the primitive integer id to a wrapper type. Using a Set would still add the key multiple times, since it's a new object reference each time.

This actually works to our benefit though, because using a primitive would still enable you to dupe handles by making him talk twice quickly. The first "finished talking" task would run, taking handles out of the `talkingHandles` list, then you could take handles out of the item frame before the 2nd "finished talking" task runs, which would replace the item frame contents and let you dupe handles.

I might've explained that really poorly, but with this setup each voice line essentially gets it's own entry in the list so you can't break handles until _all_ the voice lines are complete. Otherwise, you'd just have to wait for one to finish and it'd still be dupable.